sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -361,6 +362,27 @@ def run_pipeline(self, pipeline, options):
         experiments = list(set(experiments + debug_options.experiments))
       debug_options.experiments = experiments
 
+    # Elevate "enable_streaming_engine" to pipeline option, but using the
+    # existing experiment.
+    google_cloud_options = options.view_as(GoogleCloudOptions)
+    if google_cloud_options.enable_streaming_engine:
+      experiments = []
+      if "enable_windmill_service" not in debug_options.experiments:
+        experiments.append("enable_windmill_service")
+      if "enable_streaming_engine" not in debug_options.experiments:
+        experiments.append("enable_streaming_engine")
+      if debug_options.experiments is not None:
+        experiments = list(set(experiments + debug_options.experiments))
+      debug_options.experiments = experiments
+    else:
+      if debug_options.experiments is not None:
+        if ("enable_windmill_service" in debug_options.experiments
```

Comment:
You are checking twice for ""enable_windmill_service" in debug_options.experiments". Did you intend to check for another experiment?

Reply:
fixed, thanks